src/events/twitchStreamOnline.ts

```diff
@@ -33,38 +36,45 @@ export default class extends Event {
 			// Iterate over each subscription
 			for (const subscription of subscriptions[1]) {
 				if (subscription.status !== NotificationsStreamsTwitchEventStatus.Online) continue;
-				if (subscription.gamesBlacklist.includes(game.name) || subscription.gamesBlacklist.includes(game.id)) continue;
-				if (subscription.gamesWhitelist.length && (!subscription.gamesWhitelist.includes(game.name) || !subscription.gamesWhitelist.includes(game.id))) continue;
+				if (game !== undefined) {
+					if (subscription.gamesBlacklist.includes(game.name) || subscription.gamesBlacklist.includes(game.id)) continue;
+					if (subscription.gamesWhitelist.length && (!subscription.gamesWhitelist.includes(game.name) || !subscription.gamesWhitelist.includes(game.id))) continue;
+				}
 				if (this.client.twitch.streamNotificationDrip(`${subscriptions[0]}-${subscription.channel}-${subscription.status}`)) continue;
 
 				// Retrieve the channel, then check if it exists or if it's postable.
 				const channel = guild.channels.get(subscription.channel) as TextChannel | undefined;
 				if (typeof channel === 'undefined' || !channel.postable) continue;
 
-				// Retrieve the message and transform it, if no embed, return the basic message.
-				const message = subscription.message === null ? undefined : this.transformText(subscription.message, data, game);
-				if (subscription.embed === null) {
-					floatPromise(this, channel.send(message));
+				if (subscription.embed) {
+					// Retrieve the message and transform it into a JSON object, if the message could not be retrieved then skip this notification.
+					const embedData = subscription.message === null ? undefined : this.transformTextToObject(data, game);
+					if (embedData === undefined) break;
+
+					// Construct a message embed and send it.
+					floatPromise(this, channel.sendEmbed(this.buildEmbed(embedData, guild.language)));
 					break;
```

Comment:
This can be reduced to this:

```js
if (subscription.message !== null) {
	// Retrieve the message and transform it into a JSON object, if the message could not be retrieved then skip this notification.
	const embedData = this.transformTextToObject(data, game);

	// Construct a message embed and send it.
	floatPromise(this, channel.sendEmbed(this.buildEmbed(embedData, guild.language)));
}

break;
```